target/linux/ath79/dts/qca9531_compex_wpj531.dts

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca953x.dtsi"
+
+/ {
+	compatible = "compex,wpj531", "qca,qca9531";
+	model = "Compex WPJ531 (16MB flash)";
+
+	chosen {
+		bootargs = "console=ttyS0,115200n8";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		sig1 {
+			label = "wpj531:red:sig1";
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+		};
+
+		sig2 {
+			label = "wpj531:yellow:sig2";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+		};
+
+		sig3 {
+			label = "wpj531:green:sig3";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+		};
+
+		sig4 {
+			label = "wpj531:green:sig4";
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+		poll-interval = <30>;
+
+		reset {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&uart {
+	status = "okay";
+};
+
+&spi {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			uboot:	partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x030000>;
+				read-only;
+			};
+
+			partition@30000 {
+				label = "firmware";
+				reg = <0x030000 0xfc0000>;
```

Comment:
Adjust size.

Reply:
can you elaborate what you mean?

i've crossed-checked the sizes of the partitions here with the ar71xx config from the compex SDK `wpj531_mtdlayout_16M=mtdparts=spi0.0:192k(u-boot)ro,16128k(firmware),64k(art)ro`.

Reply:
well, earlier you said "mtdparts=ath-nor0:256k(u-boot),64k(u-boot-env),6336k(rootfs),1408k(uImage),8256k(mib0),64k(ART)" is what's set in the stock firmware.

Reply:
it's set in the default u-boot env (that is stored nowhere in the flash).
and those params are not passed to linux.
i would guess they are relics from an earlier attempt at a flash layout - as the SDK offered from compex's download section also uses the (simpler) layout above.

there are 3 more completely untouched boards in my reach.
i can check which layout is used by the default firmware - if that is of any importance. please let me now

Reply:
Well, I'd say "uboot partitioning" should be as in vendor firmware. So, if have the means to find out which that is, please do so. If you find out that ar71xx was right, then that's okay.

Reply:
i'm still waiting to get those 3 or so boards from the storage in order to be able to see what might be in the mib0-partition per factory.
in the meantime, i've installed the official compex "OSDK" (what's installed by default).

first thing to notice is that the cpximg starts at 0x30000 - so there's definitely no u-boot there.
as i've said, it appears the default u-boot-env is not stored anywhere and it is initialised dynamically on boot depending on whether the reset button was pressed or not (`bootcmd` becomes `cpximgs`).
i guess, the params in there are just an artifact of previous modifications in their u-boot fork.

the params used by the stock-kernel are:
`mtdparts=spi0.0:192k(u-boot)ro,1536k(kernel),14464k(rootfs),64k(otp)ro,64k(nvram),64k(art)ro,16000k@0x30000(firmware)`
so, the kernel and rootfs spacing is hard-coded in OSDK - but not in the official openWRT SDK.
from my point of view, the dynamic layout is superior and i see no advantage of the the hard-coded allocation.

```
$ cat /proc/mtd
dev:    size   erasesize  name
mtd0: 00030000 00010000 "u-boot"
mtd1: 00180000 00010000 "kernel"
mtd2: 00e20000 00010000 "rootfs"
mtd3: 005e0000 00010000 "rootfs_data"
mtd4: 00010000 00010000 "otp"
mtd5: 00010000 00010000 "nvram"
mtd6: 00010000 00010000 "art"
mtd7: 00fa0000 00010000 "firmware"
```

the `otp` and `nvram` both contain this:
```
admin@MimoAP:~$ hexdump -C /dev/mtd5
00000000  19 85 20 03 00 00 00 0c  f0 60 dc 98 ff ff ff ff  |.. ......`......|
00000010  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
*
00010000
```
no idea what that could be.
again, neither of those are in the official openWRT SDK and it doesn't appear they're used for anything.
there are some boards here that are missing ethernet-ports or the dc-socket - i could try whether deleting that has any impact on the functionality.

regarding the LEDs:
they're driven by the rssileds util.
i was about to get the parameters to rssileds when i burned my serial adapter and TX stopped working.
here's what it says on a WPJ344 board that it tested earlier:
`/usr/sbin/rssileds ath0 1000000 1 wpj344:red:sig1 -85 -1 0 1 wpj344:yellow:sig2 -75 -1 0 1 wpj344:green:sig3 -65 -1 0 1 wpj344:green:sig4 -55 -1 0 1`
i don't suppose it will be any different on the WPJ531 - apart from the  LED names, obviously.
however, i don't feel rssileds should become mandatory for the image.
and, again, compex' openWRT SDK doesn't use it either.